clippy_lints/src/methods/manual_position.rs

```diff
@@ -0,0 +1,180 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet_opt;
+use clippy_utils::{is_diag_item_method, is_diag_trait_item, path_to_local_id, peel_blocks_with_stmt};
+use rustc_errors::Applicability;
+use rustc_hir::{Body, ClosureKind, Expr, ExprKind, HirId, LangItem, Node, Pat, PatKind, QPath};
+use rustc_lint::LateContext;
+use rustc_span::{sym, Span};
+
+use super::MANUAL_POSITION;
+
+#[derive(Debug, PartialEq, Eq)]
+enum UsageKind {
+    Expect(Span),
+    Unwrap,
+    QuestionMark,
+    Map,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+struct Usage {
+    kind: UsageKind,
+    end_span: Span,
+}
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, arg: &Expr<'_>, start_span: Span, rev: bool) {
+    if let ExprKind::Closure(c) = arg.kind
+        && matches!(c.kind, ClosureKind::Closure)
+        && let typeck = cx.typeck_results()
+        && let Some(fn_id) = typeck.type_dependent_def_id(expr.hir_id)
+        && (is_diag_trait_item(cx, fn_id, sym::Iterator))
+        && let body = cx.tcx.hir().body(c.body)
+        && let [param] = body.params
+        && let parent = cx.tcx.hir().parent_iter(expr.hir_id)
+        && let Some(usage) = parse_usage(cx, parent)
```

Comment:
Maybe we don't need to give a "usage suggestion" at all, I mean, don't include the span of this call's parent.

for example, we can just use the span of this method call in:

```rust
let _ = seq.iter().enumerate().find(...).unwrap()
//                 ^^^^^^^^^^^^^^^^^^^^^
```

Plus, there are many way to call after an `Option`, such as `unwrap_or_default`, `unwrap_or_else`, `or_else`, `or`, etc. Parsing any missed cases will cause FN.